geomstats/learning/geometric_median.py

```diff
@@ -0,0 +1,110 @@
+"""Geometric Median Estimation."""
+
+import geomstats.backend as gs
+
+
+class WeiszfeldAlgorithm:
+    r"""Weiszfeld Algorithm for Manifolds.
+
+    Parameters
+    ----------
+    metric : RiemannianMetric
+        Riemannian metric.
+    weights : array-like, [N]
+        Weights for weighted sum.
+        Optional, default : None
+            If None equal weights (1/N) are used for all points
+    max_iter : int
+        Maximum number of iterations for the algorithm.
+        Optional, default : 100
+    lr : float
+        Learning rate to be used for the algorithm.
+        Optional, default : 1.0
+    init : array-like
+        Initialization to be used in the start.
+        Optional, default : None
+    print_every : int
+        Print updated median after print_every iterations.
+        Optional, default : None
+
+    References
+    ----------
+    .. [BJL2008]_ Bhatia, Jain, Lim. "Robust statstics on
+    Riemannian manifolds via the geometric median"
+    """
+
+    def __init__(self, metric, max_iter=100, lr=1.0, init=None, print_every=None):
+        self.metric = metric
+        self.max_iter = max_iter
+        self.lr = lr
+        self.init = init
+        self.print_every = print_every
+        self.estimate_ = None
+
+    def single_iteration(self, current_median, X, weights, lr):
```

Comment:
This should be private (add `_`  at beginning) and use a verb (e.g. `_iterate_once`?).